src/universal-dapp.js

```diff
@@ -461,103 +460,103 @@ UniversalDApp.prototype.pendingTransactions = function () {
   return this.txRunner.pendingTxs
 }
 
-UniversalDApp.prototype.runTx = function (args, cb) {
-  var self = this
-  var tx = {
-    to: args.to,
-    data: args.data,
-    useCall: args.useCall
-  }
-  async.waterfall([
-    // query gas limit
-    function (callback) {
-      tx.gasLimit = 3000000
-
-      if (self.transactionContextAPI.getGasLimit) {
-        self.transactionContextAPI.getGasLimit(function (err, ret) {
-          if (err) {
-            return callback(err)
-          }
-
-          tx.gasLimit = ret
-          callback()
-        })
-      } else {
-        callback()
-      }
-    },
-    // query value
-    function (callback) {
-      tx.value = 0
-      if (tx.useCall) return callback()
-      if (self.transactionContextAPI.getValue) {
-        self.transactionContextAPI.getValue(function (err, ret) {
-          if (err) {
-            return callback(err)
-          }
-
-          tx.value = ret
-          callback()
-        })
-      } else {
-        callback()
-      }
-    },
-    // query address
-    function (callback) {
-      if (self.transactionContextAPI.getAddress) {
-        self.transactionContextAPI.getAddress(function (err, ret) {
-          if (err) {
-            return callback(err)
-          }
-
-          tx.from = ret
+function execute (pipeline, env, callback) {
+  function next (err, env) {
+    if (err) return callback(err)
+    var step = pipeline.shift()
+    if (step) step(env, next)
+    else callback(null, env.result)
+  }
+  next(null, env)
+}
 
-          callback()
-        })
-      } else {
-        self.getAccounts(function (err, ret) {
-          if (err) {
-            return callback(err)
-          }
+UniversalDApp.prototype.replayTx = function (args, cb) {
```

Comment:
This function should do the same as `runTx` except that the pipeline should be
[queryGasLimit, runTransaction]
I m not sure the function name fit as well...

Reply:
So `replayTx` is not a good function name?

Otherwise, the function does exactly the same as `runxTx` but has the pipeline:
`[queryGasLimit, runTransaction]`, no?

What is otherwise different?

Reply:
"What is otherwise different?" : resolving of `<account -n>` .

from udapp perspektive, this is not really `replay`, not `re` at least. udapp just run transactions.

Reply:
The button in the UI is called `re-run transactions` so i think the name should be the same.
Users will anyway not see it, so it's just for us. Is that good? :-)